src/sentry/issues/producer.py

```diff
@@ -31,13 +39,14 @@ def _get_occurrence_producer() -> KafkaProducer:
 
 
 def produce_occurrence_to_kafka(
-    payload_type: PayloadType | None = PayloadType.OCCURRENCE,
+    payload_type: PayloadType = PayloadType.OCCURRENCE,
     occurrence: IssueOccurrence | None = None,
+    status_change: StatusChangeOccurrence | None = None,
     event_data: Optional[Dict[str, Any]] = None,
```

Comment:
This function is going to become pretty convoluted fast as we add more message types. It might make sense to split out the logic that is specific to a status change and then just share logic that sends the message to Kafka